### PR TITLE
[RM approach]: Adding dimensionability

### PR DESCRIPTION
**Pull request by CsatariGergely**
To provide some dimensioning guidelines this change adds dimensionability to the
list of Networking Principles & Strategy

closes: #1759

### CODE REVIEW

**Comment by walterkozlowski** [comment]
@CsatariGergely, isn't the scalability covering this?

**Comment by CsatariGergely** [comment]
> 
> 
> @CsatariGergely, isn't the scalability covering this?

The scalability part does not say that it should be possible to dimension the network in a way that it is non blocking without swapping out the leaf switches. 

@walterkozlowski I can merge these two. What do you think?

**Comment by walterkozlowski** [comment]
@CsatariGergely, I think we should merge these two.  I am afraid that "dimensionability" can create a lot of confusion as it is not a common word ...

**Comment by rabi-abdel** [comment]
@CsatariGergely would you mind please addressing @walterkozlowski input above and resolve the conflict as well (rebase)

File: doc/tech/approach.md [hunk 1]
```diff
@@ -80,27 +80,19 @@ To address the concern areas detailed above the specific topics listed below wil
 
 This section introduces some networking related principles that should be followed during development and definition of Networking scopes for Reference Model, Reference Architecture 01 (OpenStack), Reference Architecture 02 (K8s) and down to Reference Implementations and Reference Conformance test suites under CNTT.
```
Comment (pgoyal01) [inline doc/tech/approach.md:81]: for consistency in use of uppercase and lowercase as well as some suggestion on language (without a "from" use of "down to"):
```suggestion
This section introduces some networking related principles that should be followed during development and definition of the networking scope for Reference Model, Reference Architectures, Reference Implementations and Reference Conformance test suites.
```

File: doc/tech/approach.md [hunk 2]
```diff
@@ -80,27 +80,19 @@ To address the concern areas detailed above the specific topics listed below wil
...
 
-•	**Abstraction**: A standardized layer between Virtualisation Layer domains and the Networking Hardware Layer resource pool hiding the implementation details of the Networking Hardware.
+-	**Abstraction**: A standardized layer between Virtualisation Layer domains and the Networking Hardware Layer resource pool hiding the implementation details of the Networking Hardware.
```
Comment (pgoyal01) [inline doc/tech/approach.md:83]: Reduce indentation to align with other bullets in the list.
```suggestion
- **Abstraction**: A standardized layer between Virtualisation Layer domains and the Networking Hardware Layer resource pool hiding the implementation details of the Networking Hardware.
```

File: doc/tech/approach.md [hunk 3]
```diff
@@ -80,27 +80,19 @@ To address the concern areas detailed above the specific topics listed below wil
...
 > **Note:** Abstraction may be deployed in many different ways e.g. depending on target use case requirements, workload characteristics, different algorithm implementions of pipeline stages and available platforms. This includes Network Functions running on server nodes with or without programmable hardware acceleration, or networking support functions running on programmable network switches in the network.
```
Comment (pgoyal01) [inline doc/tech/approach.md:84]: Not sure how abstraction can "be deployed". "Network Functions running on server nodes with or without programmable hardware acceleration ..." is not abstraction.

What is the purpose of this Note? Maybe then we can word it appropriately.

File: doc/tech/approach.md [hunk 4]
```diff
@@ -80,27 +80,19 @@ To address the concern areas detailed above the specific topics listed below wil
...
-
-•	**Workload diversity**: Should be capable to p=to provide connectivity to any type of workloads, including VNF , CNF and BareMetal workloads.
-
-•	**Telco Grade**: Should support deployments with Telco Grade characteristics requirements.
-
-•	**Future proof**: Should support known and emerging technology trends e.g. SmartNICs, FPGAs and Programmable Switches, multi-cloud aspects (public/private/hybrid), multi-location aspects (multi-cluster, DCI, etc), Edge realted technologies, etc.
-
+- **Agnosticity** : Define Network Fabric concepts and models that can carry any type of traffic in terms of:
+  - Control, User and Management traffic types
+  - Acceleration technologies supporting multiple types of Infrastructure deployments and Network Function workloads
```
Comment (pgoyal01) [inline doc/tech/approach.md:87]: align tense with the "Agnosticity" tense:
```suggestion
  - Acceleration technologies that can support multiple types of Infrastructure deployments and Network Function workloads 
```

File: doc/tech/approach.md [hunk 5]
```diff
@@ -80,27 +80,19 @@ To address the concern areas detailed above the specific topics listed below wil
...
-•	**Telco Grade**: Should support deployments with Telco Grade characteristics requirements.
-
-•	**Future proof**: Should support known and emerging technology trends e.g. SmartNICs, FPGAs and Programmable Switches, multi-cloud aspects (public/private/hybrid), multi-location aspects (multi-cluster, DCI, etc), Edge realted technologies, etc.
-
+- **Agnosticity** : Define Network Fabric concepts and models that can carry any type of traffic in terms of:
+  - Control, User and Management traffic types
+  - Acceleration technologies supporting multiple types of Infrastructure deployments and Network Function workloads 
+-	**Automation**: Enable full automation, from Physical Fabric installation and provisioning to automation of VNF/CNF onboarding.
+-	**Openness**: All Networking is based on open source or standardized APIs (NBI and SBI) and should enable integration of Open Source networking components (e.g. SDN controllers).
+-	**Programmability**: Should enable a programmable forwarding plane controlled from a separately deployed control plane that abstracts into standardized functions and interfaces.
```
Comment (pgoyal01) [inline doc/tech/approach.md:90]: Under Agnosticity, control, user and management planes were specified and so for consistency:
```suggestion
-	**Programmability**: Should enable a programmableuser (forwarding) plane controlled from a separately deployed control plane utilising open-source/standardized functions and interfaces.
```

File: doc/tech/approach.md [hunk 6]
```diff
@@ -80,27 +80,19 @@ To address the concern areas detailed above the specific topics listed below wil
...
-•	**Workload diversity**: Should be capable to p=to provide connectivity to any type of workloads, including VNF , CNF and BareMetal workloads.
-
-•	**Telco Grade**: Should support deployments with Telco Grade characteristics requirements.
-
-•	**Future proof**: Should support known and emerging technology trends e.g. SmartNICs, FPGAs and Programmable Switches, multi-cloud aspects (public/private/hybrid), multi-location aspects (multi-cluster, DCI, etc), Edge realted technologies, etc.
-
+- **Agnosticity** : Define Network Fabric concepts and models that can carry any type of traffic in terms of:
+  - Control, User and Management traffic types
+  - Acceleration technologies supporting multiple types of Infrastructure deployments and Network Function workloads 
+-	**Automation**: Enable full automation, from Physical Fabric installation and provisioning to automation of VNF/CNF onboarding.
```
Comment (pgoyal01) [inline doc/tech/approach.md:88]: consistency
```suggestion
-	**Automation**: Enable full automation, from Physical Fabric installation and provisioning to automation of workloads (VNF/CNF) onboarding.
```

File: doc/tech/approach.md [hunk 7]
```diff
@@ -80,27 +80,19 @@ To address the concern areas detailed above the specific topics listed below wil
...
+-	**Openness**: All Networking is based on open source or standardized APIs (NBI and SBI) and should enable integration of Open Source networking components (e.g. SDN controllers).
```
Comment (pgoyal01) [inline doc/tech/approach.md:89]: nit
```suggestion
-	**Openness**: All Networking is based on open source or standardized APIs (NBI and SBI) and should enable integration of Open Source networking components such as SDN controllers.
```

File: doc/tech/approach.md [hunk 8]
```diff
@@ -80,27 +80,19 @@ To address the concern areas detailed above the specific topics listed below wil
...
+-	**Programmability**: Should enable a programmable forwarding plane controlled from a separately deployed control plane that abstracts into standardized functions and interfaces.
+-	**Scalability**: Should be scalable to handle all traffic traverse North-South and East-West enabling small up to large deployments.
+- **Workload diversity**: Should be capable to provide connectivity to any type of workloads, including VNF , CNF and BareMetal workloads.
+-	**Telco Grade**: Should support deployments with Telco Grade characteristics requirements.
```
Comment (pgoyal01) [inline doc/tech/approach.md:93]: please use either characteristics or requirements

File: doc/tech/approach.md [hunk 9]
```diff
@@ -80,27 +80,19 @@ To address the concern areas detailed above the specific topics listed below wil
...
-
-•	**Future proof**: Should support known and emerging technology trends e.g. SmartNICs, FPGAs and Programmable Switches, multi-cloud aspects (public/private/hybrid), multi-location aspects (multi-cluster, DCI, etc), Edge realted technologies, etc.
-
+- **Agnosticity** : Define Network Fabric concepts and models that can carry any type of traffic in terms of:
+  - Control, User and Management traffic types
+  - Acceleration technologies supporting multiple types of Infrastructure deployments and Network Function workloads 
+-	**Automation**: Enable full automation, from Physical Fabric installation and provisioning to automation of VNF/CNF onboarding.
+-	**Openness**: All Networking is based on open source or standardized APIs (NBI and SBI) and should enable integration of Open Source networking components (e.g. SDN controllers).
+-	**Programmability**: Should enable a programmable forwarding plane controlled from a separately deployed control plane that abstracts into standardized functions and interfaces.
+-	**Scalability**: Should be scalable to handle all traffic traverse North-South and East-West enabling small up to large deployments.
```
Comment (pgoyal01) [inline doc/tech/approach.md:91]:
```suggestion
-	**Scalability**: Should be scalable to handle all traffic traverse North-South and East-West enabling small to large deployments.
```

File: doc/tech/approach.md [hunk 10]
```diff
@@ -80,27 +80,19 @@ To address the concern areas detailed above the specific topics listed below wil
...
+- **Workload diversity**: Should be capable to provide connectivity to any type of workloads, including VNF , CNF and BareMetal workloads.
+-	**Telco Grade**: Should support deployments with Telco Grade characteristics requirements.
+-	**Future proof**: Should support known and emerging technology trends e.g. SmartNICs, FPGAs and Programmable Switches, multi-cloud aspects (public/private/hybrid), multi-location aspects (multi-cluster, DCI, etc), Edge realted technologies, etc.
+- **Dimensionability**: Should support network design in a way that the network is capable to handle the traffic in a non-blocking manner without swapping out the leaf switches.
```
Comment (pgoyal01) [inline doc/tech/approach.md:95]: "leaf switches" implies a particular networking architecture. In the RM should we keep away from such decisions.
sentence simplification
```suggestion
- **Dimensionability**: Should support networks capable of handling the traffic in a non-blocking manner without swapping out networking hardware.
```

Comment (ulikleber) [reply doc/tech/approach.md:95]: Didn't we plan to merge scalability and dimensionability? 
What about:
**Scalability** Should be scalable to handle all traffic traverse North-South and East-West enabling small to large deployments in a non-blocking manner.

File: doc/tech/approach.md (hunk 5 above)
Comment (CsatariGergely) [reply doc/tech/approach.md:90]: I agree with the user plane suggestion, but I think the utilisation of open source and/or standard interfaces is not really clear.

File: doc/tech/approach.md (hunk 3 above)
Comment (CsatariGergely) [reply doc/tech/approach.md:84]: @pgoyal01 I do not know. I did not write it. I agree that abstractions can not be deployed. What if I change it to implemented?

Comment (pgoyal01) [reply doc/tech/approach.md]: @CsatariGergely Can we first decide the purpose of this note -- what is it that we want to convey? Then we can word accordingly. Thanks

There are other terms such as "pipeline stages" (of what) that all need to be addressed.

File: doc/tech/approach.md (hunk 5 above)
Comment (pgoyal01) [reply doc/tech/approach.md]: @CsatariGergely the original wording included "standardized functions" -- I tried to generalise. I think the sentence is attempting to say that the "control" of the user plane utilises opensource and/or standadised functions and interfaces.

**Comment by CsatariGergely** [comment]
> 
> 
> @pgoyal01 and @CsatariGergely , we decided to move this contents to RM and this has been done, reviewed many times and merged. We need to raise an issue what to do with the Approach document. In my view it should be removed as it creates unnecessary discussion over an out of date contents.

Okay, I've applied the comments we discussed about them to Ch3 and removed them from the approach document to avoid further confusion.

File: doc/tech/approach.md (hunk 3 above)
Comment (CsatariGergely) [reply doc/tech/approach.md]: I'm happy to discuss this note, but this pr was supposed to be about something else. ...
For me th enote does not add any clarity. We could even remove it in a different pr.

File: doc/tech/approach.md (hunk 5 above)
Comment (CsatariGergely) [reply doc/tech/approach.md]: Yes but this wording mixes the implementation using open source and the abstraction using standardized interfaces.

**Pull request merged**